src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/helm.py

```diff
@@ -41,13 +40,12 @@ def acr_helm_show(cmd,
                   chart,
                   version=None,
                   repository='repo',
-                  resource_group_name=None,
+                  resource_group_name=None,  # pylint: disable=unused-argument
                   username=None,
                   password=None):
     login_server, username, password = get_access_credentials(
         cli_ctx=cmd.cli_ctx,
         registry_name=registry_name,
-        resource_group_name=resource_group_name,
```

Comment:
If you are altering the behavior of the command (i.e resource group used to be used but now isn't) then that would still be a breaking change even if you keep the resource-group parameter around in deprecated state. If it is deprecated, the old path should still work until it is removed.

Reply:
In the past, `resource_group` is an optional parameter.

1. If not provided, we will query it. If provided, we will use it directly to get registry. Since registry name is globally unique, resource group for a registry is also unique.
2. If the correct `resource_group` is provided, the behavior is the same as not provided.
3. If the wrong `resource_group` is provided, we error out since the resource couldn't be found.

As of this PR, `resource_group` will always be queried. So (1) and (2) will be the same. For (3), the new behavior is that the registry can still be found, rather than error out.

Reply:
We can make a one-time breaking change to remove `resource_group` also, if this change causes unacceptable side effects. What's your suggestion?

Reply:
Thanks for the clarification @djyou.  Since the behavior hasn't changed, I think you are find just deprecating for now so as not to break scripts.